File: app/src/main/java/pro/anton/averin/networking/testrest/presenters/JsonResponsePresenter.java
package pro.anton.averin.networking.testrest.presenters;

import android.os.Bundle;

import org.jetbrains.annotations.Nullable;

import javax.inject.Inject;
import javax.inject.Singleton;

import pro.anton.averin.networking.testrest.data.Storage;

@Singleton
public class JsonResponsePresenter extends BasePresenter<JsonResponseView> {

    @Inject
    Storage storage;

    @Inject
    public JsonResponsePresenter() {
    }

    @Override
    public void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getView().update(storage.getCurrentResponse());
    }
}

File: app/src/main/java/pro/anton/averin/networking/testrest/presenters/JsonResponsePresenter.kt
package pro.anton.averin.networking.testrest.presenters

import android.os.Bundle
import pro.anton.averin.networking.testrest.data.Storage
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class JsonResponsePresenter
@Inject
constructor() : BasePresenter<JsonResponseView>() {

    @Inject
    lateinit var storage: Storage

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        view.update(storage.currentResponse)
    }
}
